core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelFilterExecuterImpl.java

```diff
@@ -222,49 +224,90 @@ public BitSetGroup applyFilter(RawBlockletColumnChunks rawBlockletColumnChunks,
       }
     }
     BitSetGroup bitSetGroup = new BitSetGroup(pageNumbers);
-    for (int i = 0; i < pageNumbers; i++) {
-      BitSet set = new BitSet(numberOfRows[i]);
-      RowIntf row = new RowImpl();
-      BitSet prvBitset = null;
-      // if bitset pipe line is enabled then use rowid from previous bitset
-      // otherwise use older flow
-      if (!useBitsetPipeLine ||
-          null == rawBlockletColumnChunks.getBitSetGroup() ||
-          null == bitSetGroup.getBitSet(i) ||
-          rawBlockletColumnChunks.getBitSetGroup().getBitSet(i).isEmpty()) {
-        for (int index = 0; index < numberOfRows[i]; index++) {
-          createRow(rawBlockletColumnChunks, row, i, index);
-          Boolean rslt = false;
-          try {
-            rslt = exp.evaluate(row).getBoolean();
-          }
-          // Any invalid member while evaluation shall be ignored, system will log the
-          // error only once since all rows the evaluation happens so inorder to avoid
-          // too much log inforation only once the log will be printed.
-          catch (FilterIllegalMemberException e) {
-            FilterUtil.logError(e, false);
-          }
-          if (null != rslt && rslt) {
-            set.set(index);
+
+    if (isDimensionPresentInCurrentBlock.length == 1 && isDimensionPresentInCurrentBlock[0]) {
+      // fill default value here
+      DimColumnResolvedFilterInfo dimColumnEvaluatorInfo = dimColEvaluatorInfoList.get(0);
+      // if filter dimension is not present in the current add its default value
+      if (dimColumnEvaluatorInfo.getDimension().getDataType().isComplexType()) {
+        for (int i = 0; i < pageNumbers; i++) {
+          BitSet set = new BitSet(numberOfRows[i]);
+          RowIntf row = new RowImpl();
+          for (int index = 0; index < numberOfRows[i]; index++) {
+            ArrayQueryType complexType =
+                (ArrayQueryType) complexDimensionInfoMap.get(dimensionChunkIndex[i]);
+            int[] numberOfChild = complexType
+                .getNumberOfChild(rawBlockletColumnChunks.getDimensionRawColumnChunks(), null,
```

Comment:
how about to get all numbers of the child once

Reply:
done